models/grouped_assignments.py

```diff
@@ -238,12 +249,30 @@ def get_engagement_time(assignment, user, preclass=False, all_problem_sets=False
             # start a new session
             sessions.append(Session(activity.timestamp))
         prev = activity
+
```

Comment:
You need to add a comment describing what the earliness calculation is __supposed__ to do, so that I can make some assessment of whether it's actually working right. Ideally, this should really have a set of unit tests, and we are moving toward that for the runestone project, but it's not a requirement yet.

Reply:
Done.